src/sentry/testutils/cases.py

```diff
@@ -382,33 +382,69 @@ def __exit__(self, exc_type, exc_value, traceback):
 
 @override_settings(ROOT_URLCONF="sentry.web.urls")
 class TestCase(BaseTestCase, TestCase):
-    @pytest.fixture(autouse=True)
-    def _require_db_usage(self, monkeypatch):
+    @pytest.fixture(autouse=True, scope="class")
+    def _require_db_usage(self, request):
         # Ensure that testcases that ask for DB setup actually make use of the
         # DB. If they don't, they're wasting CI time.
+
+        class State:
+            used_db = {}
+            base = request.cls
+
+        state = State()
+
+        def finalizer():
```

Comment:
can probably execute this entire function body after the `yield` instead of registering as callback